src/polymer/call-super-in-callbacks.ts

```diff
@@ -0,0 +1,268 @@
+                  {
+                    code: this.code,
+                    severity: Severity.ERROR, sourceRange,
+                    message: stripWhitespace(`
+                  ${elementLike.className || elementLike.tagName} must call
+                  super() in its constructor because ES6 requires it.
+                `)
+                  });
+            } else {
+              const overrides = elementLike instanceof ElementMixin ?
+                  'may override a' :
+                  'overrides the';
+              warnings.push({
+                code: this.code,
+                severity: Severity.WARNING, sourceRange,
+                message: stripWhitespace(`
+                  ${getName(elementLike)} may need to call
+                  super.${methodName}() because it ${overrides}
+                  version in ${classThatRequiresSuper}.
+                `)
+              });
+            }
+          }
+        }
+      }
+    }
+    return warnings;
+  }
+}
+
+// TODO(rictic): This is awkward. Filed as
+//     https://github.com/Polymer/polymer-analyzer/issues/557
+function getParsedDocumentContaining(
+    sourceRange: SourceRange|undefined,
+    document: Document): ParsedDocument<any, any>|undefined {
+  if (!sourceRange) {
+    return undefined;
+  }
+  let mostSpecificDocument: undefined|Document = undefined;
+  for (const doc of document.getByKind('document')) {
+    if (isPositionInsideRange(sourceRange.start, doc.sourceRange)) {
+      if (!mostSpecificDocument ||
+          isPositionInsideRange(
+              doc.sourceRange!.start, mostSpecificDocument.sourceRange)) {
+        mostSpecificDocument = doc;
+      }
+    }
+  }
+  mostSpecificDocument = mostSpecificDocument || document;
+  return mostSpecificDocument.parsedDocument;
+}
+
+function getClassBody(astNode?: estree.Node|null): undefined|estree.ClassBody {
+  if (!astNode || !astNode.type) {
+    return undefined;
+  }
+  let classBody: undefined|estree.ClassBody = undefined;
+  estraverse.traverse(astNode, {
+    enter(node: estree.Node) {
+      if (node.type === 'ClassDeclaration' || node.type === 'ClassExpression') {
+        classBody = node.body;
+        return estraverse.VisitorOption.Break;
+      }
+    }
+  });
+  return classBody;
+}
+
+/**
+ * Returns the name of the class in element's inheritance chain that requires
+ * super[methodName]() be called. Returns undefined if no such class exists.
+ */
+function mustCallSuper(
+    elementLike: Element, methodName: string, document: Document): (string|
+                                                                    undefined) {
+  // TODO(rictic): look up the inheritance graph for a jsdoc tag that describes
+  //     the method as needing to be called?
+  if (!methodsThatMustCallSuper.has(methodName)) {
+    return;
+  }
+  // ElementMixins should always conditionally call super in callbacks.
+  if (elementLike instanceof ElementMixin) {
+    return `the mixin's given superclass`;
```

Comment:
I'm trying to come up with something better than `mixedin` :)

How about `a class this mixin extends`?